x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_bulk_body.test.ts

```diff
@@ -433,3 +435,197 @@ describe('buildBulkBody', () => {
     expect(fakeSignalSourceHit).toEqual(expected);
   });
 });
+
+describe('buildSignalFromSequence', () => {
+  test('builds a basic signal from a sequence of building blocks', () => {
+    const blocks = [sampleDocWithAncestors().hits.hits[0], sampleDocWithAncestors().hits.hits[0]];
+    const ruleSO = sampleRuleSO();
+    const signal = buildSignalFromSequence(blocks, ruleSO);
+    // Timestamp will potentially always be different so remove it for the test
+    // @ts-expect-error
+    delete signal['@timestamp'];
+    const expected: Omit<SignalHit, '@timestamp'> = {
```

Comment:
Optionally you can use this destructoring trick to do the omit of the property and you can avoid the `@ts-expect-error` like so:

```ts
    const ruleSO = sampleRuleSO();
    const { '@timestamp': _, ...signal } = buildSignalFromSequence(blocks, ruleSO);
    const expected: Omit<SignalHit, '@timestamp'> = {
```

<img width="739" alt="Screen Shot 2020-09-22 at 5 56 04 PM" src="https://user-images.githubusercontent.com/1151048/93949018-21126300-fcfd-11ea-97c6-995dbcff4878.png">

Reply:
You can apply that same technique to the other spots to remove the ts-expect-error as well and it keeps the code looking good.

Reply:
That's a good idea, I think we should also move the large blocks of expected values out into a reusable function so it's more clear what tweaks are being applied to the sample data in each test. For now I'm following the existing patterns to minimize the changes while adding the new EQL rule functionality, then once it's in place I definitely want to go back and refactor and reuse code better.